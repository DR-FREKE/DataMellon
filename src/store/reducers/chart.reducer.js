import {
  LOADING,
  SEND_TEST,
  ERROR_OCCURED,
  FILTER_DATA,
} from "../actions/action.types";

const initialState = {
  loading: false,
  error: false,
  filtering: false,
};

const postChartReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOADING:
      return {
        ...state,
        loading: true,
      };
    case SEND_TEST:
      return {
        ...state,
        loading: false,
        filtering: false,
        data: action.payload.data,
      };
    case FILTER_DATA:
      return {
        ...state,
        loading: false,
        filtering: true,
        filtered_data: action.payload.data,
      };
    case ERROR_OCCURED:
      return {
        ...state,
        loading: false,
        error: true,
        error_msg: action.payload,
      };
    default:
      return state;
  }
};

export default postChartReducer;
