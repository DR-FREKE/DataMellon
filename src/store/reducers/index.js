export { default as uiReducer } from "./ui.reducer";
export { default as chartReducer } from "./chart.reducer";
