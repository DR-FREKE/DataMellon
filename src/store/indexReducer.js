import { combineReducers } from "redux";
import * as reducer from "./reducers";

const reducers = combineReducers({
  mode: "development",
  ui: reducer.uiReducer,
  chart: reducer.chartReducer,
});

export default reducers;
