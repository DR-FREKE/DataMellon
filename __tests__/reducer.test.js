import { chartReducer, uiReducer } from "../src/store/reducers";
import * as actions from "../src/store/actions/action.creator";

const initialState = {
  dark_mode: false,
};

describe("test the reducer", () => {
  it("should test for app mode", () => {
    expect(
      uiReducer(initialState, actions.appMode({ theme: false }))
    ).toMatchSnapshot();
  });

  it("should test for default state", () => {
    expect(uiReducer({}, actions.appMode)).toMatchSnapshot();
  });

  it("should test for chartReducer", () => {
    expect(chartReducer({}, actions.postAction)).toMatchSnapshot();
  });

  it("should return initial state for chartReducer", () => {
    expect(chartReducer({ loading: true }, actions.postAction)).toEqual({
      loading: true,
    });
  });
});
