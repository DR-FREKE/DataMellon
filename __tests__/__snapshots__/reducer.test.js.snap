// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the reducer should test for chartReducer 1`] = `Object {}`;
